libraries/core_libs/consensus/src/final_chain/rewards_stats.cpp

```diff
@@ -42,35 +44,46 @@ std::optional<addr_t> RewardsStats::getTransactionValidator(const trx_hash_t& tx
   return {found_tx->second};
 }
 
-bool RewardsStats::addVote(const Vote& vote) {
+bool RewardsStats::addVote(const std::shared_ptr<Vote>& vote, uint64_t min_weight) {
   // Set valid cert vote to validator
-  auto& validator_stats = validators_stats_[vote.getVoterAddr()];
-  assert(validator_stats.valid_cert_vote_ == false);
+  auto& validator_stats = validators_stats_[vote->getVoterAddr()];
+  assert(validator_stats.vote_weight_ == 0);
+  assert(vote->getWeight());
+
+  if (validator_stats.vote_weight_) {
+    return false;
+  }
 
-  validator_stats.valid_cert_vote_ = true;
-  total_unique_votes_count_++;
+  const auto weight = *vote->getWeight();
+  if (total_votes_weight_ > min_weight) {
```

Comment:
let' say there are 4 votes with weight 1 and min_weight == 3, bonus_votes_weight_ will be 0 in such scenario but it should be 1. There should be `>=`